File: src/main/java/com/shockang/study/algorithm/java/leetcode/leetcode1_100/leetcode1/solution1/Solution.java
package com.shockang.study.algorithm.java.leetcode.leetcode1_100.leetcode1.solution1;

import java.util.HashMap;
import java.util.Map;

/**
 * 1. 两数之和
 * <p>
 * 哈希表
 *
 * @author Shockang
 */
public class Solution {
	public int[] twoSum(int[] nums, int target) {
		int n = nums.length;
		Map<Integer, Integer> map = new HashMap<>(n);
		for (int i = 0; i < n; i++) {
			if (map.containsKey(nums[i])) {
				return new int[]{i, map.get(nums[i])};
			} else {
				map.put(target - nums[i], i);
			}
		}
		return null;
	}
}

File: src/main/scala/com/shockang/study/algorithm/scala/leetcode/leetcode1_100/leetcode1/solution1/Solution.scala
package com.shockang.study.algorithm.scala.leetcode.leetcode1_100.leetcode1.solution1

import scala.collection.mutable

/**
 * 1. 两数之和
 * <p>
 * 哈希表
 *
 * @author Shockang
 */
object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    val map = mutable.HashMap[Int, Int]()
    for (i <- nums.indices)
      if (map.contains(nums(i)))
        return Array(map(nums(i)), i)
      else
        map(target - nums(i)) = i
    Array(-1, -1)
  }
}
